app/src/main/java/com/framgia/soundclound/data/source/local/sqlite/TrackLocalDataSource.java

```diff
@@ -0,0 +1,55 @@
+package com.framgia.soundclound.data.source.local.sqlite;
+
+import android.content.Context;
+
+import com.framgia.soundclound.data.model.Album;
+import com.framgia.soundclound.data.model.Track;
+import com.framgia.soundclound.data.source.TrackDataSource;
+import com.framgia.soundclound.util.Constant;
+
+import java.util.List;
+
+/**
+ * Created by Bui Danh Nam on 10/1/2018.
+ */
+
+public class TrackLocalDataSource implements TrackDataSource.LocalDataSource {
+    private static TrackLocalDataSource INSTANCE;
```

Comment:
WARNING: [StaticFieldLeak] Do not place Android context classes in static fields (static reference to `TrackLocalDataSource` which has field `mContext` pointing to `Context`); this is a memory leak (and also breaks Instant Run)
 A static field will leak contexts.

Non-static inner classes have an implicit reference to their outer class. If that outer class is for example a `Fragment` or `Activity`, then this reference means that the long-running handler/loader/task will hold a reference to the activity which prevents it from getting garbage collected.

Similarly, direct field references to activities and fragments from these longer running instances can cause leaks.

ViewModel classes should never point to Views or non-application Contexts.